lib/vsc/install/shared_setup.py

```diff
@@ -1254,13 +1275,17 @@ def parse_target(self, target, urltemplate=None):
 
         # update the cmdclass with ones from vsc_setup_klass
         # cannot do this in one go, when SHARED_TARGET is defined, vsc_setup doesn't exist yet
-        for name, klass in new_target['cmdclass'].items():
+        keepers = new_target.copy()
+        for name in list(new_target['cmdclass']):
```

Comment:
@vsoch The `list` is no longer necessary now that you're using `keepers`, is it?

Reply:
I think it is - we have to iterate through something (it can't be keepers if we are making changes) and creating a list in this fashion is the way to have support for python2 and 3.

Reply:
But of course there are likely other solutions that would work here if you have preference!

Reply:
I still think it's fine to simply drop the `list` part...

We're iterating of the keys of a dictionary here, which we are not modifying, that works the same in both Python 2 & 3? Or am I missing something?

Reply:
I just un did the change, you will see the error I believe.

Reply:
```python
======================================================================
ERROR: test_parse_target (test.shared_setup.TestSetup)
Test for parse target
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/vanessa/Documents/Dropbox/Code/Python/easybuild/vsc-install/test/shared_setup.py", line 193, in test_parse_target
    new_target = setup.parse_target(package)
  File "lib/vsc/install/shared_setup.py", line 1279, in parse_target
    for name in new_target['cmdclass']:
RuntimeError: dictionary changed size during iteration
```

Reply:
I just pushed the fix again. Let me know how you'd like to proceed for the args/ kwargs string checking! The changes I had originally done adjust the regular expression to handle the different orders (and not fail tests).

Reply:
Ah, I think you should `.copy` `new_target['cmdclass']` rather than `new_target` itself (or take a `deepcopy` of the entire `new_target`)...